workflow-pacta-cop.Rmd

```diff
@@ -0,0 +1,516 @@
+
+
+
+```
+
+
+
+
+### Checking the peer group files
+
+Run through the following chunk to check that the files you've created make sense. Make adjustments as necessary. 
+
+```{r echo = FALSE}
+
+# checking the peer group files
+
+# Here is a list of the peer groups being checked for. If your project contains different ones, enter them here. 
+peer_groups <- c("pensionfund","insurance","bank","assetmanager")
+peer_groups <- c("pensionfund", "bank")
+
+# Add an estimated minimum number of investors you have in the study
+number_of_investors = 4
+
+if (file.exists(file.path(data_path, paste0(project_code, "_peers_equity_results_portfolio.rda")))){
+  peers_equity_results_portfolio <- read_rds(file.path(data_path, paste0(project_code, "_peers_equity_results_portfolio.rda")))
+  
+  if (length(setdiff(unique(peers_equity_results_portfolio$portfolio_name), c(peer_groups, "Meta Portfolio"))) > 0){
+    print("Unexpected occurances in peers_equity_results_portfolio. The following is missing from the portfolio_name col:")
+    print(setdiff(unique(peers_equity_results_portfolio$portfolio_name), c(peer_groups, "Meta Portfolio")))
+  } 
+  
+  if (length(setdiff(unique(peers_equity_results_portfolio$investor_name), c(peer_groups, "Meta Investor"))) > 0){
+    print("Unexpected occurances in peers_equity_results_portfolio. The following is missing from the investor_name col:")
+    print(setdiff(unique(peers_equity_results_portfolio$investor_name),  c(peer_groups, "Meta Investor")))
+  } 
+  
+}else{
+  print("No Equity peer group results")
+}
+
+if(file.exists(file.path(data_path, paste0(project_code, "_peers_bonds_results_portfolio.rda")))){
+  peers_bonds_results_portfolio <- read_rds(file.path(data_path, paste0(project_code, "_peers_bonds_results_portfolio.rda")))
+  
+  if (length(setdiff(unique(peers_bonds_results_portfolio$portfolio_name), c(peer_groups, "Meta Portfolio"))) > 0){
+    print("Unexpected occurances in peers_bonds_results_portfolio. The following is missing from the portfolio_name col:")
+    print(setdiff(unique(peers_bonds_results_portfolio$portfolio_name),  c(peer_groups, "Meta Portfolio")))
+  } 
+  
+  if (length(setdiff(unique(peers_bonds_results_portfolio$investor_name), c(peer_groups, "Meta Investor"))) > 0){
+    print("Unexpected occurances in peers_bonds_results_portfolio. The following is missing from the investor_name col:")
+    print(setdiff(unique(peers_bonds_results_portfolio$investor_name), c(peer_groups, "Meta Investor")))
+  } 
+  
+}else{
+  print("No Bond peer group results")
+}
+
+if (file.exists(file.path(data_path, paste0(project_code, "_peers_equity_results_portfolio_ind.rda")))){
+  peers_equity_results_user <- read_rds(file.path(data_path, paste0(project_code, "_peers_equity_results_portfolio_ind.rda")))
```

Comment:
surprised @maurolepore has mentioned DRY yet

may want to set the filepath to a variable first, then test if it exists, then read it, like...
``` r
path_ peers_equity_results_user <- file.path(data_path, paste0(project_code, "_peers_equity_results_portfolio_ind.rda"))
if (file.exists(path_ peers_equity_results_user)) {
  peers_equity_results_user <- read_rds(path_ peers_equity_results_user)
}
```

helps with readability/understandability